File: kotlin/PlanesAndroid/app/src/main/java/com/planes/common/GuessPoint.java
package com.planes.common;

public class GuessPoint implements Cloneable {

    public GuessPoint(int row, int col) {
        m_row = row;
        m_col = col;
    }

    public GuessPoint(int row, int col, Type tp) {
        m_row = row;
        m_col = col;
        m_type = tp;
    }

    //sets the result of the guess
    public void setType(Type tp) {
        m_type = tp;
    }

    @Override
    public boolean equals(final Object other) {
        if (this == other)
            return true;
        if (other == null)
            return false;
        if (getClass() != other.getClass())
            return false;
        GuessPoint gp = (GuessPoint) other;

        return (m_row == gp.m_row) && (m_col == gp.m_col);
    }

    public boolean isDead() {
        return m_type == Type.Dead;
    }
    public boolean isHit() {
        return m_type == Type.Hit;
    }
    public boolean isMiss() {
        return m_type == Type.Miss;
    }

    public Object clone() {
        return new GuessPoint(m_row, m_col, m_type);
    }

    public int m_row = 0;
    public int m_col = 0;
    public Type m_type = Type.Miss;
}

File: kotlin/PlanesAndroid/app/src/main/java/com/planes_multiplayer/single_player_engine/GuessPoint.kt
package com.planes_multiplayer.single_player_engine

class GuessPoint(val row: Int, val col: Int, val tp: Type) : Cloneable {

    var m_row = 0
    var m_col = 0
    var m_type = Type.Miss

    init {
        m_row = row
        m_col = col
        m_type = tp
    }

    constructor(row: Int, col: Int): this(row, col, Type.Miss) {
        m_row = row
        m_col = col
    }

    fun row(): Int {
        return m_row
    }

    fun col(): Int {
        return m_col
    }

    fun type(): Type {
        return m_type
    }
    //sets the result of the guess
    fun setType(tp: Type) {
        m_type = tp
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null) return false
        if (javaClass != other.javaClass) return false
        val gp = other as GuessPoint
        return m_row == gp.m_row && m_col == gp.m_col
    }

    val isDead: Boolean
        get() = m_type === Type.Dead
    val isHit: Boolean
        get() = m_type === Type.Hit
    val isMiss: Boolean
        get() = m_type === Type.Miss

    public override fun clone(): Any {
        return GuessPoint(m_row, m_col, m_type)
    }
}